File: lectures/code/cc_03_01.cpp
#include <iostream>
#include <math.h>

class Point {
    public: 

    double x, y;
    Point(double xc, double yc)  {
        x = xc;
        y = yc;
    };

    void dump() {
        printf("Object point x=%f y=%f\n", x, y);
    }

    double origin() {
        return sqrt(x*x+y*y);
    }
};

int main() {
  Point pt(4.0, 5.0);
  pt.dump();
  std::cout << "Origin: " << pt.origin() << "\n";
  return 0;
}

// rm -f a.out; g++ cc_03_01.cpp; a.out; rm -f a.out

File: lectures/code/cc_03_01.js
function Point(x, y) {
    this.x = x;
    this.y = y;
    this.party = function () {
        this.x = this.x + 1;
        console.log("So far "+this.x);
    }

    this.dump = function () {
        console.log("Object point x=%f y=%f", this.x, this.y);
    }

    this.origin = function () {
        return Math.sqrt(this.x*this.x+this.y*this.y);
    }
}

pt = new Point(4.0, 5.0);

pt.dump();
console.log("Origin %f",pt.origin());

// node cc_03_01.js
